pkg/didcomm/protocol/didexchange/persistence.go

```diff
@@ -86,14 +99,60 @@ func (c *ConnectionRecorder) GetInvitation(verKey string) (*Invitation, error) {
 	return result, nil
 }
 
-// invitationKey computes key for invitation object
-func invitationKey(verKey string) (string, error) {
-	storeKey, err := computeHash([]byte(verKey))
+// SaveConnectionRecorder saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) SaveConnectionRecorder(invitation *Invitation,
+	msg *Message) error {
+	connRecord := &connectionRecord{ThreadID: msg.ThreadID,
+		ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey:          invitation.RecipientKeys, State: msg.State}
+
+	connBytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	return c.store.Put(msg.ConnectionID, connBytes)
```

Comment:
Use key prefix, 
(rebase to latest code, follow `SaveInvitation()`)

Reply:
done